numba/npyufunc/dufunc.py

```diff
@@ -52,6 +63,10 @@ def __init__(self, py_func, **kws):
             kws.pop('identity', None))
         super(DUFunc, self).__init__(dispatcher, **kws)
         self._install_type()
+        self._kernel = type('DUFuncKernel_' + self.ufunc.__name__,
```

Comment:
Coming from a C/C++ perspective, I find declaring a class inside a function to be confusing.  I also find changing the class name after the class definition to be clumsy.